src/nix/search.cc

```diff
@@ -20,16 +21,52 @@ std::string wrap(std::string prefix, std::string s)
     return prefix + s + ANSI_NORMAL;
 }
 
+#define HILITE_COLOR ANSI_GREEN
+
 std::string hilite(const std::string & s, const std::smatch & m, std::string postfix)
 {
     return
         m.empty()
         ? s
         : std::string(m.prefix())
-          + ANSI_GREEN + std::string(m.str()) + postfix
+          + HILITE_COLOR + std::string(m.str()) + postfix
           + std::string(m.suffix());
 }
 
+std::string hilite_all(const std::string &s, std::vector<std::smatch> matches, std::string postfix) {
```

Comment:
That function starts being non-trivial. It’s probably worth moving into `libutil`, and add a few tests for it

Reply:
I'm certainly going to add tests.
I think moving the function into libutil makes sense but I'm not very familiar with the nix source code, should I put it in a new file or for example `fmt.hh`.

Reply:
I’d say either `fmt.hh` or `ansicolor.hh` would make sense. Tbh libutil would need some restructuring because it has grown a bit messy over time

Reply:
I moved it into `fmt.hh` and created an `fmt.cc` to avoid linker errors, this required changing the signature to accept a vector but I hope that's fine.